rules/cache/cache.go

```diff
@@ -224,9 +234,12 @@ func (c *cache) setWithLock(
 	if invalidate {
 		results = c.invalidateWithLock(nsHash, idHash, results)
 	}
-	res := results.source.Match(id, c.nowFn())
+	now := c.nowFn()
+	earliest := now.Add(-c.maxNegativeSkew)
+	latest := now.Add(c.maxPositiveSkew)
+	res := results.source.Match(id, earliest, latest)
 	newElem := &element{nsHash: nsHash, idHash: idHash, result: res}
-	newElem.SetExpiry(c.newExpiry(c.nowFn()))
+	newElem.SetExpiry(c.newExpiry(now))
```

Comment:
No, the expiry time here is used to control when the items will be promoted, essentially we don't update the elements position on every access to avoid the locking overhead of promotion and instead promote (i.e., moving element to the front of the list) every X minutes, so the expiry time here should be determined based on current local time.

I can perhaps change `SetExpiry` and `newExpiry` to `SetPromotionExpiry` and `newPromotionExpiry` to avoid confusion between this expiry time and the expiry time of the match result.